### DISCUSSION TITLE
I got an error when working with "rpart2"

### DISCUSSION THREAD

**Discussion by sskoldas** [SchlossLab/mikropml#311] [Q&A] [answered]
Hello!
I was working on "glmnet, rf, svm, rpart2" with my otu dataset. 
While "glmnet, rf, svm" models were running without an error, I got an error for "rpart2" model. I checked my script again, I did exactly what was written in the instruction. But, I got the same error again.

I attached a screenshot of the error that I got, below. I didn't understand the error. I also added code that I used for "rpart2".
Where did I make mistake? 
Thank you

`test_hp <- list(maxdepth= c(1, 2,4,8,16,30)) 
get_response_genus_results <- function(seed){
  run_ml(response_genus_preprocess,
       method = "rpart2",
       outcome_colname = "response_diff",
       kfold = 5, 
       cv_times = 100,
       training_frac = 0.8, 
       hyperparameters = test_hp,
       seed = seed)
}`
![Screenshot from 2022-11-07 20-36-56](https://user-images.githubusercontent.com/55446435/200377560-2aed7c76-83e4-4130-b75e-b8074bc6b442.png)

Those are also warnings:
Warning messages:
1: Setting row names on a tibble is deprecated.
2: predictions failed for Fold1.Rep001: maxdepth=30 Error in approx(x[, "nsplit"], x[, "CP"], depth) : 
  need at least two non-NA values to interpolate

3: Setting row names on a tibble is deprecated.
4: Setting row names on a tibble is deprecated.
5: predictions failed for Fold3.Rep001: maxdepth=30 Error in approx(x[, "nsplit"], x[, "CP"], depth) : 
  need at least two non-NA values to interpolate

6: Setting row names on a tibble is deprecated.
7: Setting row names on a tibble is deprecated.
8: predictions failed for Fold5.Rep001: maxdepth=30 Error in approx(x[, "nsplit"], x[, "CP"], depth) : 
  need at least two non-NA values to interpolate

9: Setting row names on a tibble is deprecated.
10: predictions failed for Fold1.Rep002: maxdepth=30 Error in approx(x[, "nsplit"], x[, "CP"], depth) : 
  need at least two non-NA values to interpolate

11: Setting row names on a tibble is deprecated.
12: predictions failed for Fold2.Rep002: maxdepth=30 Error in approx(x[, "nsplit"], x[, "CP"], depth) : 
  need at least two non-NA values to interpolate

13: Setting row names on a tibble is deprecated.
14: Setting row names on a tibble is deprecated.
15: Setting row names on a tibble is deprecated.
16: predictions failed for Fold5.Rep002: maxdepth=30 Error in approx(x[, "nsplit"], x[, "CP"], depth) : 
  need at least two non-NA values to interpolate

17: Setting row names on a tibble is deprecated.
18: predictions failed for Fold1.Rep003: maxdepth=30 Error in approx(x[, "nsplit"], x[, "CP"], depth) : 
  need at least two non-NA values to interpolate

19: Setting row names on a tibble is deprecated.
20: Setting row names on a tibble is deprecated.
21: Setting row names on a tibble is deprecated.
22: predictions failed for Fold4.Rep003: maxdepth=30 Error in approx(x[, "nsplit"], x[, "CP"], depth) : 
  need at least two non-NA values to interpolate

23: Setting row names on a tibble is deprecated.
24: Setting row names on a tibble is deprecated.
25: predictions failed for Fold1.Rep004: maxdepth=30 Error in approx(x[, "nsplit"], x[, "CP"], depth) : 
  need at least two non-NA values to interpolate

26: Setting row names on a tibble is deprecated.
27: predictions failed for Fold2.Rep004: maxdepth=30 Error in approx(x[, "nsplit"], x[, "CP"], depth) : 
  need at least two non-NA values to interpolate

28: Setting row names on a tibble is deprecated.
29: Setting row names on a tibble is deprecated.
30: Setting row names on a tibble is deprecated.
31: predictions failed for Fold5.Rep004: maxdepth=30 Error in approx(x[, "nsplit"], x[, "CP"], depth) : 
  need at least two non-NA values to interpolate

32: Setting row names on a tibble is deprecated.
33: predictions failed for Fold1.Rep005: maxdepth=30 Error in approx(x[, "nsplit"], x[, "CP"], depth) : 
  need at least two non-NA values to interpolate

34: Setting row names on a tibble is deprecated.
35: Setting row names on a tibble is deprecated.
36: predictions failed for Fold3.Rep005: maxdepth=30 Error in approx(x[, "nsplit"], x[, "CP"], depth) : 
  need at least two non-NA values to interpolate

37: Setting row names on a tibble is deprecated.
38: predictions failed for Fold4.Rep005: maxdepth=30 Error in approx(x[, "nsplit"], x[, "CP"], depth) : 
  need at least two non-NA values to interpolate

39: Setting row names on a tibble is deprecated.
40: predictions failed for Fold5.Rep005: maxdepth=30 Error in approx(x[, "nsplit"], x[, "CP"], depth) : 
  need at least two non-NA values to interpolate

41: Setting row names on a tibble is deprecated.
42: Setting row names on a tibble is deprecated.
43: Setting row names on a tibble is deprecated.
44: Setting row names on a tibble is deprecated.
45: predictions failed for Fold4.Rep006: maxdepth=30 Error in approx(x[, "nsplit"], x[, "CP"], depth) : 
  need at least two non-NA values to interpolate

46: Setting row names on a tibble is deprecated.
47: predictions failed for Fold5.Rep006: maxdepth=30 Error in approx(x[, "nsplit"], x[, "CP"], depth) : 
  need at least two non-NA values to interpolate

48: Setting row names on a tibble is deprecated.
49: Setting row names on a tibble is deprecated.
50: Setting row names on a tibble is deprecated.

**Comment by kelly-sovacool** [comment, marked as answer]
For `need at least two non-NA values to interpolate`: I've seen this warning for very small or imbalanced datasets. It appears that the model does so poorly it gets `NA` values for predictions during training. 

For `Setting row names on a tibble is deprecated.`: somewhere in the caret or rpart2 code it attempts to set row names, which is no longer supported for tibbles. It should be harmless.
